idaes/models_extra/gas_distribution/parsers/tests/test_networkdata_model.py

```diff
@@ -0,0 +1,140 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Tests the pyomo model created by the NetworkData parser
+"""
+
+import pytest
+import os
+import unittest
+from pyomo.common.fileutils import this_file_dir
+from urllib.request import urlopen
+from zipfile import ZipFile
+from io import BytesIO
+from socket import timeout
+from idaes.models_extra.gas_distribution.parsers.gaslib_parser import parse_gaslib_network
+
+pytest.importorskip("plotly", reason="plotly not available")
+
+try:
+    data = urlopen("https://gaslib.zib.de/download/testData/GasLib-11.zip", timeout=3)
+    zipfile = ZipFile(BytesIO(data.read()))
+    files = [zipfile.open(file) for file in zipfile.namelist()]
+    netfile = files[1]
+    scnfile = files[2]
+    net = parse_gaslib_network(netfile, scnfile)
+    m = net.build_model(with_bounds=False)
+
+except timeout:
+    raise unittest.SkipTest('An internet connection is required to test gaslib parsing tools')
```

Comment:
This also seems like useful functionality that we might want to import and use elsewhere. Maybe a function like the following could live in `gaslib_parser.py`?
```python
def get_gaslib_data(number=11, tout=3):
    if number != 11: 
        raise ValueError("Only GasLib-11 is supported for now.")
    url = "https://gaslib.zib.de/download/testData/GasLib-%s.zip" % int(number)
    try:
        data = urlopen(url, timeout=tout)
    except timeout:
        raise RuntimeError(
            "An internet connection is required to use gaslib parsing tools"
        )
    zipfile = ZipFile(BytesIO(data.read()))
    files = [zipfile.open(file) for file in zipfile.namelist()]
    netfiles = [f for f in files if f.name.endswith(".net")]
    scnfiles = [f for f in files if f.name.endswith(".scn")]
    if len(netfiles) > 1:
        raise RuntimeError(
            "More than one network file encountered when processing %s" % url 
        )
    if len(scnfiles) > 1:
        raise RuntimeError(
            "More than one scenario file encountered when processing %s" % url 
        )
    netfile = netfiles[0]
    scnfile = scnfiles[0]
    net = parse_gaslib_network(netfile, scnfile)
    return net 
```
Note that I tested this with the other GasLib test networks and got errors in the parser for all networks other than GasLib-11. Does this match your experience?

Reply:
Note that the gaslib link here appears to be broken. The link that works for me now is:
```console
https://gaslib.zib.de/download/testData/GasLib-11-v1-20211130.zip
```
Maybe this is a good argument for not making this a publicly available function...

Reply:
Hey Robbie sorry I didn’t see these comments before I left Sandia it was a bit hectic of a time… The code snippet for getting gaslib data seems to say only gaslib11 is supported. I believe Michael wrote this so I’d check with him to see if it’s been updated since.